crypto/encrypt.go

```diff
@@ -0,0 +1,101 @@
+package crypto
+
+import (
+	"crypto/cipher"
+	"crypto/rand"
+	"errors"
+
+	"golang.org/x/crypto/twofish"
+)
+
+const (
+	BlockSize = twofish.BlockSize
+	KeySize   = 32
+)
+
+type (
+	EncryptionKey [KeySize]byte
+)
+
+// Encryption and decryption of []bytes is supported, but requires keeping some
+// overhead information. Also, nothing is done in-place, which means the
+// functions are more memory intensive, but that's not generally the bottleneck
+// when doing encryption. Developers are less likely to make mistakes when
+// inputs and outputs are different memory, which is why this method has been
+// chosen. Additionally, the overhead of using encryption is more transparent.
+
+// GenerateEncryptionKey produces a key that can be used for encrypting and
+// decrypting files.
+func GenerateEncryptionKey() (key EncryptionKey, err error) {
+	_, err = rand.Read(key[:])
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+// EncryptBytes encrypts a []byte using a key. The padded ciphertext, iv, and
+// amount of padding used are returned. `plaintext` is not overwritten.
+func EncryptBytes(key EncryptionKey, plaintext []byte) (ciphertext []byte, iv []byte, padding int, err error) {
+	// Determine the length needed for padding. The ciphertext must be padded
+	// to a multiple of BlockSize.
+	padding = BlockSize - (len(plaintext) % BlockSize)
+	if padding == BlockSize {
+		padding = 0
+	}
```

Comment:
this took a second for me to parse. I might have written it:
```go
padding := 0
if rem := len(plaintext) % BlockSize; rem != 0 {
    padding = BlockSize - rem
}
```
but neither way is clearly better than the other.